primitives/election-providers/src/lib.rs

```diff
@@ -160,23 +163,43 @@
 
 pub mod onchain;
 use sp_std::{prelude::*, fmt::Debug};
+use data_provider::ReturnValue;
 
 /// Re-export some type as they are used in the interface.
 pub use sp_arithmetic::PerThing;
 pub use sp_npos_elections::{Assignment, ExtendedBalance, PerThing128, Supports, VoteWeight};
 
+pub mod data_provider {
+	/// Alias for the return value of the election data provider.
+	pub type ReturnValue<T, Aux> = Result<(T, Aux)>;
+	/// Alias for the result type of the election data provider.
+	pub type Result<T> = sp_std::result::Result<T, &'static str>;
+}
+
 /// Something that can provide the data to an [`ElectionProvider`].
 pub trait ElectionDataProvider<AccountId, BlockNumber> {
+	/// Additional data can optionally be returned with the data. An example is an indication of the
+	/// resourced consumed during the operation.
+	type Additional: Default;
```

Comment:
Why do you bound Default ?

Reply:
Some reason that no longer exists afaik. Removed.